internal/component/input/async_reader.go

```diff
@@ -122,6 +122,11 @@ func (r *AsyncReader) loop() {
 				r.mgr.Logger().Errorf("Failed to connect to %v: %v\n", r.typeStr, err)
 				mFailedConn.Incr(1)
 
+				var e component.ErrBackOff
+				if errors.As(err, &e) {
+					_ = r.sleepWithCancellation(closeAtLeisureCtx, e.Wait)
```

Comment:
@Jeffail Interested in your thoughts on this. Initially I was thinking a time.Sleep would be sufficient but it'd be good to be able to cancel it to which I found this solution which is a bit complicates things, perhaps unnecessarily.